From 44f0e71756c680cc709a020bb604d53d6c1a6704 Mon Sep 17 00:00:00 2001
From: Benjamin Franzke <bfr@qbus.de>
Date: Thu, 13 Apr 2017 00:26:58 +0200
Subject: [PATCH] Debug rawCommandLineArguments

v2: use output buffering and print to stderr, to
not confuse typo3_console when reading stdout

Change-Id: I2d004cb3b0b896a0753f12bb43cfe775b3154a7f
---
 typo3/sysext/extbase/Classes/Mvc/Cli/RequestBuilder.php | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/typo3/sysext/extbase/Classes/Mvc/Cli/RequestBuilder.php b/typo3/sysext/extbase/Classes/Mvc/Cli/RequestBuilder.php
index 27e9719..53d0bf5 100644
--- a/typo3/sysext/extbase/Classes/Mvc/Cli/RequestBuilder.php
+++ b/typo3/sysext/extbase/Classes/Mvc/Cli/RequestBuilder.php
@@ -90,6 +90,12 @@ class RequestBuilder implements \TYPO3\CMS\Core\SingletonInterface
         $request->setCallingScript($callingScript);
         $request->setControllerObjectName(\TYPO3\CMS\Extbase\Command\HelpCommandController::class);
         $rawCommandLineArguments = is_array($commandLine) ? $commandLine : explode(' ', $commandLine);
+
+        ob_start();
+        var_dump($rawCommandLineArguments);
+        $debug = ob_get_clean();
+        fwrite(STDERR, $debug);
+
         if (empty($rawCommandLineArguments)) {
             $request->setControllerCommandName('helpStub');
             return $request;
-- 
2.9.3

